### DISCUSSION TITLE
About eigenstrain in RSPHERICAL coord_type

### DISCUSSION THREAD

**Discussion by Xfengrui** [idaholab/moose#21475] [Q&A Modules: Solid mechanics] [answered]
Hi,

I'm trying to use coord_type = RSPHERICAL to solve a 1D hollow sphere problem.
The mesh info is 
```
[Mesh]
    type = GeneratedMesh
    dim = 1
    xmin = 2e-3
    xmax = 3e-3
    nx = 1
[]
```
I wish the the thickness of the sphere shell increase from 1mm to 1.1mm. So I gave a eigenstrain of '0.1 0.1 .0.1' as follows:
```
  [./eigen]
    type = ComputeEigenstrain
    eigen_base = '0.1 0.1 0.1'
    eigenstrain_name = eigenstrain
    outputs = exodus
  [../]
```
However, the thickness of the shell after the calculation became 1.2mm rather than 1.1mm.

I then give the same eigenstrain to a solid sphere with r=1e-3, and the radius of the sphere increased to 1.1mm as expected.  I'm now wondering if I gave a wrong eigenstrain for the hollow sphere in view of the coord_type = RSPHERICAL  ?
```
[Mesh]
    type = GeneratedMesh
    dim = 1
    xmin = 0    #solid sphere with r=1e-3
    xmax = 1e-3
    nx = 1
[]
```

**Comment by GiudGiud** [comment]
Hello

Well what is the analytical result for this?

Also only one element in the mesh? Are you supposed to get the exact solution with just one?

Guillaume

**Reply by Xfengrui** [reply]
Thanks for your reply. 

In sphere coord type, strain_rr = du_rr / dr, thus the analytical result for this should be about 1.1mm from my point of view. And the solid sphere case has proved it.

I also have tried to use more elements but it didn't change the results.

**Reply by GiudGiud** [reply]
What is the spatial extent of the 1D hollow shell after you apply the strain? Like the new xmin and xmax?

**Reply by Xfengrui** [reply]
I gave a fixed boundary condition for "left", so the new xmin = 2e-3  and xmax= 3.2e-3. 
For the solid sphere, the new xmin = 0 and xmax = 1.1e-3.

**Reply by GiudGiud** [reply]
I think strain for a shell would be dr / r so since r is 2e-3, the dr should be 0.1 * 2e-3 not 0.1 * 1e-3
found a cool video about it
https://www.youtube.com/watch?v=7t9t4Sotl8U

**Reply by Xfengrui** [reply]
Sorry, I'm not sure I have understood it. 
r=2e-3 for inner radius, r=3e-3 for outer radius. In the case that inner radius is fixed,  dr shoud be 3e-4 for the outer?

**Reply by GiudGiud** [reply]
I think the formula takes the inner radius for r

**Reply by Xfengrui** [reply]
Thanks for your help! I think you are correct in the case of free BCs. And the elongation should be (0.1*3e-3-0.1*2e-3)=0.1e-3.

Here I imposed a Dirichlet BC for inner side and the problem seem to be much more complicated according to @hugary1995 .

**Comment by hugary1995** [comment, marked as answer]
This is actually an interesting question (plus it's boring here at the airport...), so I decided to write it down:

The strain measure we have in tensor_mechanics for large deformation is notionally the logarithmic strain
![image](https://user-images.githubusercontent.com/31555204/176988491-029bdd98-1af7-4c47-b537-8e1293f7f3e5.png)
where
![image](https://user-images.githubusercontent.com/31555204/176988477-cb3de577-1d30-45d2-8ac3-622955f327ec.png)

Say the r-displacement is u, then the total strain is 
![image](https://user-images.githubusercontent.com/31555204/176988545-6c06dbde-9c2c-49a2-8e30-3dc19f6d028f.png)

Subtracting the diagonal eigenstrain you get the mechanical strain as
![image](https://user-images.githubusercontent.com/31555204/176988614-c1cc9b2f-aa86-430e-b877-fcc44c1429fe.png)

Let Young's modulus be E, and poisson's ratio be 0 to make my life easier. The stress is
![image](https://user-images.githubusercontent.com/31555204/176988660-76c4d79e-b1ba-4afb-8601-fa7cc184b0a2.png)

Next, the linear momentum balance is
![image](https://user-images.githubusercontent.com/31555204/176988709-dc7061cb-d340-45a2-8f32-53a7372df53e.png)
which has become a second order nonlinear ordinary differential equation that I don't know how to solve analytically. So I'll stop here. But I think this is enough to convince you that the problem isn't as simple as it looks like.

A couple of remarks here:
- To get what you want, you'll need to make many simplifications. If you make the small strain assumption, and use Poisson's ratio = 0, and do not impose _any_ Dirichlet boundary conditions, the elongation will be exactly the eigenstrain times the thickness of the hollow sphere, spot on.
- If the Poisson's ratio gets larger, the elongation will be larger due to z and theta contributions, which is the so-called Poisson's effect.
- If you are additionally imposing a Dirichlet BC on the inner surface, you are effectively applying a traction on that surface, which will make the solution further away from your guess.

It seems to me that you doing all of the above: large deformation, using a nonzero Poisson's ratio, and imposing a Dirichlet BC, which makes the problem very difficult (analytically, not numerically). (Although I know it seems to be pretty straightforward...) Not to mention the fact that we are doing a specific form of objective integration, and using a specific hypoelasticity model -- all of that complicates the picture.

**Reply by hugary1995** [reply]
Oh, and a minor correction after inspecting the math, if you use small strain, and remove all the Dirichlet BCs, then the Poisson's ratio actually won't affect the result in this specific case.

For example, if you have a domain [2e-3, 3e-3], then using `strain = SMALL` in the TM action, (or `ComputeRSphericalSmallStrain`), and don't impose any Dirichet BCs, the displacements are always gonna be 2e-4 and 3e-4 at the left and the right nodes, with an eigenstrain of 0.1, regardless of the value of the Poisson's ratio.

**Reply by Xfengrui** [reply]
Thanks a lot!
I moved the DirichletBCs  and the elongation obtained was 0.1mm as I expected. And I think that the elongation was much larger than 0.1 mm before is mainly due to the "traction" as you said.  Thanks for your helpful suggestions.
However, here comes another problem. The case I mentioned above is somewhat simplified. In a real situation, the sphere have a three -layers structure, and the growth occurs at the middle layer (which means that the two side of the layer are both affected by the other layers). To get a exact elongation of the middle layer will be tricky then. 
What's more, it's not very clear for me about whether the eigenstrain models in Cartesian coordinates (for example, thermal expansions) are still suitable for Spherical coordinates for hollow sphere cases in view that we will get different elongations.
